core/primitives/src/crypto/signer.rs

```diff
@@ -198,36 +204,62 @@ impl InMemorySigner {
         let mut rng = OsRng::new().expect("Unable to generate random numbers");
         let account_id: String =
             rng.sample_iter(&Alphanumeric).filter(|c| c.is_ascii_alphabetic()).take(10).collect();
-        let (public_key, secret_key) = signature::get_key_pair();
+        let (public_key, secret_key) = get_key_pair();
         let bls_secret_key = BlsSecretKey::generate();
         let bls_public_key = bls_secret_key.get_public_key();
         Self { account_id, public_key, secret_key, bls_public_key, bls_secret_key }
     }
 }
 
+// Use hash for the Key so that we do not store raw crypto keys in the memory. Also more memory
+// efficient.
+cached_key!{
+  TRANSACTION_CACHE: SizedCache<CryptoHash, Signature> = SizedCache::with_size(1_000_000);
+  Key = {
+        let mut res = sk.to_bytes();
+        res.extend_from_slice(data);
+        hash(&res)
+        };
+  fn sign_transaction(data: &[u8], sk: &SecretKey) -> Signature  = {
+    sign(data, sk)
+  }
+}
+
 impl TransactionSigner for InMemorySigner {
     #[inline]
     fn public_key(&self) -> PublicKey {
         self.public_key
     }
 
-    fn sign(&self, data: &[u8]) -> signature::Signature {
-        signature::sign(data, &self.secret_key)
+    fn sign(&self, data: &[u8]) -> Signature {
+        sign_transaction(data, &self.secret_key)
     }
 }
 
+cached_key!{
+  BLOCK_CACHE: SizedCache<CryptoHash, PartialSignature> = SizedCache::with_size(100_000);
+  Key = {
```

Comment:
Is there a way to make the Signer pointer the key so we don't have to compute the hash?

Reply:
That would require some refactoring to wrap signer into `Pin<>` to guarantee that its address does not change. Also we probably want signers to share LRU, if we have different signers running on different shards, or shard and beacon. So I am not sure we want that.

Reply:
there is no way to with this library to cache a class member?

Reply:
By class member you mean `InMemorySigner`? Then we would cache the secret key with it which we wanted to avoid.

Reply:
I mean make BLOCK_CACHE a member of InMemorySigner and map from <hash we're signing, signature>

Reply:
Addressed in another comment.